component/src/main/java/io/siddhi/extension/map/protobuf/sinkmapper/ProtobufSinkMapper.java

```diff
@@ -0,0 +1,407 @@
+                ),
+                @Example(
+                        syntax = "@sink(type='grpc', url = 'grpc://localhost:2000/org.wso2.grpc.test" +
+                                ".MyService/testMap' \n" +
+                                "@map(type='protobuf')) \n" +
+                                " define stream BarStream (stringValue string,intValue int,map object);",
+
+                        description = "The above definition will map BarStream values to request message type of the " +
+                                "'testMap' method in 'MyService' service and since there is an object data type is in" +
+                                " " +
+                                "the stream(map object) , mapper will assume that 'map' is an instance of  " +
+                                "'java.util.Map' class, otherwise it will throws and error. \n" +
+                                ""
+                )
+        }
+)
+
+public class ProtobufSinkMapper extends SinkMapper {
+
+    private Class builder;
+    private Class messageObjectClass;
+    private Object messageBuilderObject;
+    private List<MappingPositionData> mappingPositionDataList;
+    private String siddhiAppName;
+
+
+    /**
+     * Returns a list of supported dynamic options (that means for each event value of the option can change) by
+     * the transport
+     *
+     * @return the list of supported dynamic option keys
+     */
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    /**
+     * The initialization method for {@link SinkMapper}, which will be called before other methods and validate
+     * the all configuration and getting the initial values.
+     *
+     * @param streamDefinition containing stream definition bind to the {@link SinkMapper}
+     * @param optionHolder     Option holder containing static and dynamic configuration related
+     *                         to the {@link SinkMapper}
+     * @param map              Unmapped payload for reference
+     * @param configReader     to read the sink related system configuration.
+     * @param siddhiAppContext the context of the {@link io.siddhi.query.api.SiddhiApp} used to
+     *                         get siddhi related utilty functions.
+     */
+    @Override
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, Map<String, TemplateBuilder> map,
+                     ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+        String[] attributeNameArray = streamDefinition.getAttributeNameArray();
+        this.siddhiAppName = siddhiAppContext.getName();
+        if (sinkType.equals(GrpcConstants.GRPC_SERVICE_RESPONSE_SINK_NAME) && map.size() == 0) {
+            throw new SiddhiAppCreationException(siddhiAppContext.getName() + "no mapping found at @map, mapping " +
+                    "should be available to continue"); //grpc-service-source should have a mapping
```

Comment:
Change the error message something like "No mapping found at @map, mapping should be available to continue for Siddhi App + siddhiAppContext.getName()